experimental/dds/tree2/src/shared-tree-core/editManager.ts

```diff
@@ -530,22 +573,54 @@ export class EditManager<
 			this.trunkUndoRedoManager.trackCommit(trunkHead, type);
 		}
 		this.trunk.repairDataStoreProvider?.applyChange(commit.change);
-		this.sequenceMap.set(sequenceNumber, trunkHead);
-		this.trunkMetadata.set(trunkHead.revision, { sequenceNumber, sessionId: commit.sessionId });
+		this.sequenceMap.set(sequenceId, trunkHead);
+		this.trunkMetadata.set(trunkHead.revision, { sequenceId, sessionId: commit.sessionId });
 		this.events.emit("newTrunkHead", trunkHead);
 	}
 
 	/**
-	 * Finds the most recent trunk commit that was sequenced at or before the given sequence number.
+	 * Finds the most recent trunk commit that was sequenced at or before the given sequence number. If the commit was part of a batch an index can be pass.
 	 * @param sequenceNumber - the sequence number to search for
+	 * @param indexInBatch - the commit index inside the batch to search for in conjunction with the sequence number
 	 * @remarks Fails if there is no eligible commit.
-	 * @returns the closest commit and its sequence number
+	 * @returns the closest commit and its sequence id
 	 */
-	private getClosestTrunkCommit(sequenceNumber: SeqNumber): [SeqNumber, GraphCommit<TChangeset>] {
-		const commit = this.sequenceMap.getPairOrNextLower(sequenceNumber);
-		assert(commit !== undefined, 0x72a /* Sequence number has been evicted */);
+	private getClosestTrunkCommit(
+		sequenceNumber: SeqNumber,
+		indexInBatch?: number,
+	): [SequenceId, GraphCommit<TChangeset>] {
+		const sequenceId: SequenceId =
+			indexInBatch !== undefined
+				? {
+						sequenceNumber,
+						indexInBatch,
+				  }
+				: {
+						// This is to make sure that if the commit index in the batch is not passed the correct commit is selected in this 2 scenarios:
+						// 1) The commit is unique for that sequence number
+						// 2) There are more than one commit for the same sequence number, in this case we need to select the last one.
+						sequenceNumber: brand((sequenceNumber as number) + 1),
```

Comment:
Isn't this equivalent to doing `{ sequenceNumber, indexInBatch: Number.Infinity }` like we do in `trimTrunk()`?